tools/perf/ResultsComparer/src/ResultsComparer/VsProfiler/VsProfilerReader.cs

```diff
@@ -0,0 +1,227 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="VsProfilerReader.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using ResultsComparer.Core;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace ResultsComparer.VsProfiler
+{
+    // TODO: Consider using a general-purpose CSV/TSV reader instead

```

Comment:
Do you need this TODO yet a TextReader is being used?

Reply:
I realized that the reports for VS profilers are actually TSV, so here I am pretty much implementing a custom TSV reader. I think we can benefit from using a library that's built for that. In future iterations, it would be great if we can support comparing two arbitrary CSV files that have measurement data. Using a general purpose csv reader would make that task easier and would be able to handle edge cases and nuances that come with CSV files (e.g. escaping)